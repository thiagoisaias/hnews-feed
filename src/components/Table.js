import React from "react";
import TableItem from "./TableItem";
import styled from "styled-components";
import PropTypes from "prop-types";

const Container = styled.div`
  padding: 16px;
  background-color: #f6f6ef;
  margin-top: 80px;
`;

const Table = props => {
  return (
    <Container>
      {props.list.map(item => (
        <TableItem key={item.objectID} onDismiss={props.onDismiss} {...item} />
      ))}
    </Container>
  );
};

Table.propTypes = {
  list: PropTypes.array.isRequired,
  onDismiss: PropTypes.func.isRequired
};

export default Table;
