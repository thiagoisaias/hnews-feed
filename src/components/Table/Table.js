import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import TableItem from "../TableItem/TableItem";

const Container = styled.div`
  padding: 16px;
  background-color: #f6f6ef;
  margin-top: 130px;
`;

const Table = props => {
  const { list, page } = props;
  return (
    <Container>
      {list.map(item => <TableItem key={item.objectID + page} {...item} />)}
    </Container>
  );
};

Table.propTypes = {
  list: PropTypes.array.isRequired
};

export default Table;
