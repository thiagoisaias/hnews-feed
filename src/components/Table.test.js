import React from "react";
import Enzyme, { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Table from "./Table";

describe("Table", () => {
  it("renders without crashing", () => {
    const listForTest = [
      {
        objectID: 1,
        url: "random.url.test",
        title: "random item test",
        points: 553.43
      },
      {
        objectID: 2,
        url: "random.url.test.2",
        title: "random item 2 test ",
        points: 253.43
      }
    ];

    const functionForTest = () => {
      return;
    };

    const wrapper = shallow(
      <Table list={listForTest} onDismiss={functionForTest} />
    );
  });

  it("has a valid snapshot", () => {
    const listForTest = [
      {
        objectID: 1,
        url: "random.url.test",
        title: "random item test",
        points: 553.43
      },
      {
        objectID: 2,
        url: "random.url.test.2",
        title: "random item 2 test ",
        points: 253.43
      }
    ];

    const functionForTest = () => {
      return;
    };

    const component = shallow(
      <Table list={listForTest} onDismiss={functionForTest} />
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
