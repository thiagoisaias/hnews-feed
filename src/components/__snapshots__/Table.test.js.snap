// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table has a valid snapshot 1`] = `
<styled.div>
  <TableItem
    author="Author Lorem"
    created_at="2017-05-05T14:14:17.000Z"
    key="1"
    objectID={1}
    points={553.43}
    title="random item test"
    url="random.url.test"
  />
  <TableItem
    author="Author Lorem"
    created_at="2017-05-05T14:14:17.000Z"
    key="2"
    objectID={2}
    points={253.43}
    title="random item 2 test "
    url="random.url.test.2"
  />
</styled.div>
`;
